File: app/src/main/java/com/dominikgaller/alpinebooking/guide/core/domain/guidetour/exception/TourStartTooEarlyException.java
package com.dominikgaller.alpinebooking.guide.core.domain.guidetour.exception;

import java.time.Instant;

/**
 * Thrown when a guide tour start is attempted before the scheduled start time.
 *
 * <p>Maps to HTTP 409 Conflict at the REST boundary.
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}, section 8.
 */
public class TourStartTooEarlyException extends RuntimeException {

    public TourStartTooEarlyException(final Instant scheduledStart, final Instant attemptedAt) {
        super("Tour cannot be started before scheduled start time. Scheduled: "
                + scheduledStart + ", attempted: " + attemptedAt);
    }
}

File: src/main/kotlin/com/example/contractmanagement/guide/core/domain/guidetour/exception/TourStartTooEarlyException.kt
package com.example.contractmanagement.guide.core.domain.guidetour.exception

import java.time.Instant

/**
 * Thrown when a guide tries to start a tour before its scheduled time. Maps to 409.
 *
 * A conflict rather than a bad request: the input is well-formed, the world is not ready.
 */
class TourStartTooEarlyException(
    scheduledStart: Instant,
    attemptedStart: Instant,
) : RuntimeException("Tour is scheduled for $scheduledStart and cannot start at $attemptedStart")
